generate.sh

```diff
@@ -88,6 +88,14 @@ ENVVARS="$(python ${BASE}/modules/scripts/config.py "${USERNAME}" "${REPONAME}")
 echo -e "\n${ENVVARS}\n" >> ${LOGFILE}
 eval $ENVVARS
 
+if [ -n "$TARGET_BRANCH" ]; then
+  git checkout "$TARGET_BRANCH" >> ${LOGFILE}
```

Comment:
Currently I'm reading the branch from `.yaydoc.yml`. I only know the branch after it's cloned. That's why i had to use checkout. However upon reconsideration, I think it would be better if `branch` comes directly from the webUI and not from `.yaydoc.yml`(like how travis does it). That way we'll just use `.yaydoc.yml` from whatever branch is choosen from the webUI and we can even have different configurations for different branches. This would also mean that while registering with ci, users would also have to specify a branch.

@imujjwal96 @sch00lb0y What do you suggest?